README.md

```diff
@@ -1,2 +1,86 @@
+
+    **Example:** Stored locally as `./params.yml`
+
+    ```yaml
+    input: s3://example-bucket/input.csv
+    synapse_config: s3://example-bucket/synapse_config.ini
+    outdir: s3://example-bucket/synapse/
+    ```
+
+4. Launch workflow using the [Nextflow CLI](https://nextflow.io/docs/latest/cli.html#run), the [Tower CLI](https://help.tower.nf/latest/cli/), or the [Tower web UI](https://help.tower.nf/latest/launch/launchpad/).
+
+    **Example:** Launched using the Tower CLI
+
+    ```console
+    tw launch sage-bionetworks-workflows/nf-synstage --params-file=./params.yml
+    ```
+
+5. Retrieve the output file. The Synapse URIs have been replaced with their staged locations. This file can now be used as the input for other workflows.
+
+    **Example:** Downloaded from `s3://example-bucket/synapse/<run-name>/input.staged.csv`
+
+    ```text
+    sample,fastq_1,fastq_2,strandedness
+    foobar,s3://example-bucket/synapse/syn28521174/foobar.R1.fastq.gz,s3://example-bucket/synapse/syn28521175/foobar.R2.fastq.gz,unstranded
+    ```
+
+## Authentication
+
+Downloading files from Synapse requires the workflow to be authenticated. The workflow currently supports two authentication methods:
+
+- **(Preferred)** Create a secret called `SYNAPSE_AUTH_TOKEN` containing a Synapse personal access token using the [Nextflow CLI](https://nextflow.io/docs/latest/secrets.html) or [Nextflow Tower](https://help.tower.nf/latest/secrets/overview/). 
+- Provide a Synapse configuration file containing a personal access token (see example above) to the `synapse_config` parameter. This method is best used if Nextflow/Tower secrets aren't supported on your platform.
+
+You can generate a personal access token using [this dashboard](https://www.synapse.org/#!PersonalAccessTokens:).
+
+## Parameters
+
+Check out the [Quickstart](#quickstart) section for example parameter values.
+
+- **`input`**: A text file containing Synapse URIs (_e.g._ `syn://syn28521174`). The text file can have any format (_e.g._ a single column of Synapse URIs, a CSV/TSV sample sheet for an nf-core workflow).
+
+- **`synapse_config`**: (Optional) A [Synapse configuration file](https://python-docs.synapse.org/build/html/Credentials.html#use-synapseconfig) containing authentication credentials. A minimal example is included in the [Quickstart](#quickstart) section.
+
+- **`outdir`**: An output location where the Synapse files will be staged. Currently, this location must be an S3 prefix.
+
+## Known Limitations
+
+- The workflow only supports S3 buckets as target staging locations.
+- The only way for the workflow to download Synapse files is by listing Synapse URIs in a file. You cannot provide a list of Synapse IDs or URIs to a parameter.
+- The workflow doesn't check if newer versions exist for the files associated with the Synapse URIs. If you need to force-download a newer version, you should manually delete the staged version.
```

Comment:
Because Nextflow caches tasks based on inputs and other aspects. If the `synapse_get` process gets the input of `syn12345678` and downloads the file, the next time it gets that same input (_i.e._ `syn12345678`), it will think see that it already ran with that input and use the cached version. I hope to make it smarter with #3.